core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -2559,6 +2559,18 @@ public void testDynamicStarInTableJoin() throws Exception {
     sql(sql).with(getTesterWithDynamicTable()).ok();
   }
 
+  /**
+   * Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-2900">[CALCITE-2900]
+   * RewriteRexShuttle generates wrong type during flatten for structured type when
+   * visitInputRef</a>.
+   */
+  @Test
+  public void testNestedColumn() {
+    final String sql =
+        "select HOME_ADDRESS.ZIP from SALES.EMP_ADDRESS where HOME_ADDRESS.CITY = 'abc'";
```

Comment:
Minor: I guess it is just a matter of style but since the majority of tests use lowercase formatting for table and column identifiers maybe it is worth doing the same. Of-course it is not consistent all over the place so I have no strong preference on this.

Reply:
Good call. Done.